src/ashley/views.py

```diff
@@ -118,6 +112,44 @@ def _do_on_login(self, lti_request: LTI) -> HttpResponse:
 
         return response
 
+    @classmethod
+    def _check_marked_locked_unlocked_unsync(cls, forum, context):
+        """
+        The context contains a field is_marked_locked, we control that if
+        this field is checked or unchecked, there is no problem of unsynchronicity
+        It's a security to remove/add forum permission on the default group.
+        This could only happen if the course was locked/unlocked outside
+        web navigation directly on the database
+        """
+        # forum is marked locked but the base group still has all the writing
+        # permissions
+        locked_unsync = context.is_marked_locked and all(
```

Comment:
From what I understand, I don't think you're checking the right thing. Do you agree with the following statements or am I missing something?
- if if is marked locked then a synchronization issue would be that ANY write permission exists
- If it is marked unlocked, then a synchronization issue would be that NOT ALL write permissions are given

Reply:
If I'm right then there must also be missing tests to make sure it works correctly:
- To prove that the synchronization check for a locked forum works: create a locked forum and add 1 random write permission > assert that the check fails
- To prove that the synchronization check for an unlocked forum works: create an unlocked forum, add all write permissions except 1 taken randomly  > assert that the check fails

Reply:
I've changed the behaviour regarding your last feedback. I'm going to change it back the way I've submitted it first to go back on my first idea.
I find it dangerous to consider a synchronization issue when 
* If it is marked unlocked, then a synchronization issue would be that NOT ALL write permissions are given
Permissions in some how could be changed outside this context, what is sure is that if none of the writing permissions are assigned whereas it's unlocked, it has to be a synchronization issue.

The same way if any write permissions exists I agee and it was this way first, there we could consider a synchronization issue.

Reply:
There's something I don't get: who assigns these write permissions? My understand is that it's done programmatically and that it's all or nothing. Is it not the case?

Reply:
yes but it could be changed in the administration